big_tests/tests/mod_event_pusher_rabbit_SUITE.erl

```diff
@@ -0,0 +1,321 @@
+              %% THEN exchanges are deleted
+              [?assertException(_Error, _Reason,
+                                error_when_exchange_is_absent(Connection,
+                                                              Channel, Exchange,
+                                                              ?IF_EXCHANGE_EXISTS_RETRIES))
+               || Exchange <- Exchanges]
+      end).
+
+presence_messages_are_properly_formatted(Config) ->
+    escalus:story(
+      Config, [{bob, 1}],
+      fun(Bob) ->
+              %% GIVEN
+              escalus:wait_for_stanzas(Bob, 1),
+              BobJID = nick_to_jid(bob, Config),
+              BobRoutingKey = presence_rk(BobJID),
+              listen_to_presence_events_from_rabbit([BobJID], Config),
+              %% WHEN user logout
+              escalus:send(Bob, escalus_stanza:presence(<<"unavailable">>)),
+              %% THEN receive message
+              receive
+                  {#'basic.deliver'{routing_key = BobRoutingKey},
+                   #amqp_msg{payload = BobMsg}} ->
+                      ?assertMatch(#{<<"user_id">> := BobJID,
+                                     <<"present">> := false},
+                                   jiffy:decode(BobMsg, [return_maps]))
+              after
+                  5000 -> exit(timeout)
+              end
+      end).
+%%--------------------------------------------------------------------
+%% Test helpers
+%%--------------------------------------------------------------------
+
+-spec listen_to_rabbit() -> proplists:proplist().
+listen_to_rabbit() ->
+    {ok, Connection} =
+        amqp_connection:start(#amqp_params_network{host = "localhost"}),
+    {ok, Channel} = amqp_connection:open_channel(Connection),
+    [{rabbit_connection, Connection}, {rabbit_channel, Channel}].
+
+-spec close_rabbit_connection(Config :: proplists:proplist()) -> ok | term().
+close_rabbit_connection(Config) ->
+    Connection = proplists:get_value(rabbit_connection, Config),
+    Channel = proplists:get_value(rabbit_channel, Config),
+    amqp_channel:close(Channel),
+    amqp_connection:close(Connection).
+
+-spec listen_to_presence_events_from_rabbit(JIDs :: [binary()],
+                                            Config :: proplists:proplist()) ->
+    ok | term().
+listen_to_presence_events_from_rabbit(JIDs, Config) ->
+    Connection = proplists:get_value(rabbit_connection, Config),
+    Channel = proplists:get_value(rabbit_channel, Config),
+    Queue = declare_rabbit_queue(Channel),
+    QueueBindings = presence_bindings(Queue, JIDs),
+    wait_for_exchanges_to_be_created(Connection, JIDs),
+    bind_queues_to_exchanges(Channel, QueueBindings),
+    subscribe_to_rabbit_queue(Channel, Queue).
+
+-spec wait_for_exchanges_to_be_created(Connection :: pid(),
+                                       Exchanges :: [binary()]) -> pid().
+wait_for_exchanges_to_be_created(Connection, Exchanges) ->
```

Comment:
You may add similiar function `wait_for_exchanges_to_be_deleted`.

Reply:
It could be used in test `disconnected_users_delete_exchanges_when_unavailable(Config)`